k2/csrc/array.h

```diff
@@ -112,21 +105,20 @@ struct Array2 {
   }
 
   IndexT size1;
+  IndexT size2;     // the number of elements in the array,  equal to
```

Comment:
The initialization order of data members in the initialization list of the constructors should match the definition order.

Reply:
Actually I prefer to keep the original order (size1, indexes, size2, data) here as it's more readable (indexes is related to size1, and data is related to data).

Reply:
Different people have different opinions about what is more readable.

The current approach is more memory-efficient.

Reply:
Will merge and leave it with `unresolved`. We can update it later if other guys (@danpovey) think we should change the order in the initialization list